Library/Homebrew/extend/os/mac/hardware/cpu.rb

```diff
@@ -63,6 +63,14 @@ def universal_archs
         [arch_64_bit, arch_32_bit].extend ArchitectureListExtension
       end
 
+      # True when running under an Intel-based shell via Rosetta on an
+      # Apple Silicon Mac. This can be detected via seeing if there's a
+      # conflict between what `uname` report and the underlying `sysctl` flags,
+      # since the `sysctl` flags don't change behaviour under Rosetta.
+      def running_under_rosetta?
+        Utils.popen_read("uname", "-m").chomp == "x86_64" && arm64?
```

Comment:
Why not just `sysctl_bool("sysctl.proc_translated")`?

Reply:
(From https://developer.apple.com/documentation/apple_silicon/about_the_rosetta_translation_environment)

Reply:
Returns an empty/false value under both Intel and ARM. It may be that this is a check that would work via `sysctl` syscalls but not via the commandline tool.

Reply:
Hmm, yeah, I see the same. Probably worth filing a radar for, since it is the documented way to check this.

Reply:
Yep, I've filed a Feedback. We'll see what we hear.

Reply:
```suggestion
        native_type != type
```

Reply:
```suggestion
        native_arm64? && !arm64?
```